Microsoft.Toolkit.Win32/Microsoft.Toolkit.Wpf.UI.XamlHost/WindowsXamlHostBase.cs

```diff
@@ -0,0 +1,187 @@
+            }
+
+            set
+            {
+                if (value == ChildInternal)
+                {
+                    return;
+                }
+
+                var currentRoot = (Windows.UI.Xaml.FrameworkElement)ChildInternal;
+                if (currentRoot != null)
+                {
+                    currentRoot.SizeChanged -= XamlContentSizeChanged;
+                }
+
+                SetContent(value);
+
+                var frameworkElement = ChildInternal as Windows.UI.Xaml.FrameworkElement;
+                if (frameworkElement != null)
+                {
+                    // If XAML content has changed, check XAML size
+                    // to determine if WindowsXamlHost needs to re-run layout.
+                    frameworkElement.SizeChanged += XamlContentSizeChanged;
+
+                    // WindowsXamlHost DataContext should flow through to UWP XAML content
+                    frameworkElement.DataContext = DataContext;
+                }
+
+                // Fire updated event
+                ChildChanged?.Invoke(this, new EventArgs());
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this wrapper control instance been disposed
+        /// </summary>
+        protected bool IsDisposed { get; set; }
+
+        /// <summary>
+        /// Creates <see cref="Windows.UI.Xaml.Application" /> object, wrapped <see cref="Windows.UI.Xaml.Hosting.DesktopWindowXamlSource" /> instance; creates and
+        /// sets root UWP XAML element on <see cref="Windows.UI.Xaml.Hosting.DesktopWindowXamlSource" />.
+        /// </summary>
+        /// <param name="hwndParent">Parent window handle</param>
+        /// <returns>Handle to XAML window</returns>
+        protected override HandleRef BuildWindowCore(HandleRef hwndParent)
+        {
+            // 'EnableMouseInPointer' is called by the WindowsXamlManager during initialization. No need
+            // to call it directly here.
+
+            // Create DesktopWindowXamlSource instance
+            var desktopWindowXamlSourceNative = _xamlSource.GetInterop();
+
+            // Associate the window where UWP XAML will display content
+            desktopWindowXamlSourceNative.AttachToWindow(hwndParent.Handle);
+
+            var windowHandle = desktopWindowXamlSourceNative.WindowHandle;
+
+            // Overridden function must return window handle of new target window (DesktopWindowXamlSource's Window)
+            return new HandleRef(this, windowHandle);
+        }
+
+        /// <summary>
+        /// The default implementation of SetContent applies ChildInternal to desktopWindowXamSource.Content.
+        /// Override this method if that shouldn't be the case.
+        /// For example, override if your control should be a child of another WindowsXamlHostBase-based control.
+        /// </summary>
+        protected virtual void SetContent(Windows.UI.Xaml.UIElement newValue)
```

Comment:
With this exposed, the WPF designer does not load. Consumers would need to write their own code to load the WinRT type. See previous comments about the default implementation using now `ChildInternal` property. Without a solution here, #2427 is blocked

Reply:
I have a solution for this in #2432 that I'll move here

Reply:
Copied the solution for this in.  @ryalanms please take a look, also, just to double check?